website/public/js/controllers/challengesCtrl.js

```diff
@@ -314,6 +316,28 @@ habitrpg.controller("ChallengesCtrl", ['$rootScope','$scope', 'Shared', 'User',
       $scope.search.group = _.transform($scope.groups, function(m,g){m[g._id] = false});
     }
 
+    $scope.shouldShow = function(task, list, prefs){
+      return true;
```

Comment:
Ahh, I see. You accidentally duplicated this from below, and the function below overwrites the shouldShow function from the tasks controller in the view when used with the challenges controller.